app/src/main/java/org/sil/hearthis/RecordActivity.java

```diff
@@ -160,27 +179,74 @@ void nextButtonClicked() {
 		scrollView.scrollTo(0, scrollPos);
 	}
 	
-	void recordButtonTouch(MotionEvent e) {
+	void recordButtonTouch(MotionEvent e)
+	{
 		int maskedAction = e.getActionMasked();
 
-	    switch (maskedAction) {
-		    case MotionEvent.ACTION_DOWN: {
-		    	startRecording();
-		      break;
-		    }
-		    case MotionEvent.ACTION_UP:
-		    case MotionEvent.ACTION_CANCEL: {
-		    	stopRecording();
-		      break;
-		    }
-	    }
+		switch (maskedAction)
+		{
+		case MotionEvent.ACTION_DOWN:
+			record.setBackgroundColor(0xffff0000);
+			startRecording();
+			break;
+
+		case MotionEvent.ACTION_UP:
+		case MotionEvent.ACTION_CANCEL:
+			record.setBackgroundColor(0x3f000000);
+			stopRecording();
+			break;
+		}
+	}
+
+	@Override
+	public boolean onKeyDown(int keyCode, KeyEvent event)
+	{
+		switch (keyCode)
+		{
+		case KeyEvent.KEYCODE_VOLUME_DOWN:
+		case KeyEvent.KEYCODE_VOLUME_UP:
+			record.setBackgroundColor(0xffff0000);
+			startRecording();
+			return true;
+		default:
+			return onKeyDown(keyCode, event);
+		}
 	}
 
-	String _fileName = "";
+	@Override
+	public boolean onKeyUp(int keyCode, KeyEvent event)
+	{
+		switch (keyCode)
+		{
+		case KeyEvent.KEYCODE_VOLUME_DOWN:
+		case KeyEvent.KEYCODE_VOLUME_UP:
+			record.setBackgroundColor(0x3f000000);
+			stopRecording();
+			return true;
+		default:
+			return super.onKeyUp(keyCode, event);
+		}
+	}
 
-	void startRecording() {
-		if (recorder != null) {
-	      recorder.release();
+	@Override
+	public boolean onKeyLongPress(int keyCode, KeyEvent event)
+	{
+		switch (keyCode)
+		{
+		case KeyEvent.KEYCODE_VOLUME_DOWN:
+		case KeyEvent.KEYCODE_VOLUME_UP:
+			// ignore the long press, as the button is being held
+			return true;
+		default:
+			return super.onKeyUp(keyCode, event);
```

Comment:
It's odd to call the super version of a different method. If you really meant to do that, I think a comment indicating why is called for.